File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int current_size = nums.size();
        std::vector<int>ans(current_size * 2);
        for (int idx = 0; idx < current_size; idx++) {
            ans[idx] = nums[idx];
            ans[idx + current_size] = nums[idx];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sort-an-array/submission-1.rs
impl Solution {
    pub fn sort_array(nums: Vec<i32>) -> Vec<i32> {
        let mut arr = nums.clone();
        insertion_sort(&mut arr);
        arr
    }
}

fn insertion_sort(arr: &mut [i32]) {
    (1..arr.len()).for_each(|idx| {
        let key = arr[idx];
        let mut j = idx as isize - 1;
        while j >= 0 && key < arr[j as usize] {
            arr[(j + 1) as usize] = arr[j as usize];
            j -= 1;
        }
        arr[(j + 1) as usize] = key;
    });
}


